### PR TITLE
fix(pointcloud preprocessor): fix ring outlier filter and concatenation to use PointXYZIRC 

### PR DESCRIPTION
**Pull request by a-maumau**
## Description

The current implementation of [RingOutlierFilter](https://github.com/autowarefoundation/autoware.universe/blob/5a6cde954790dc744b2ec10e88749437e8b57f9e/sensing/pointcloud_preprocessor/src/outlier_filter/ring_outlier_filter_nodelet.cpp#L27C1-L27C27) and [PointCloudConcatenateDataSynchronizer](https://github.com/autowarefoundation/autoware.universe/blob/main/sensing/pointcloud_preprocessor/src/concatenate_data/concatenate_and_time_sync_nodelet.cpp) does not follow the [documentation of the Autoware](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-architecture/sensing/data-types/point-cloud/#recommended-processing-pipeline).

This PR will change the output of RingOutlierFilter and PointCloudConcatenateDataSynchronizer from `PointXYZI` to `PointXYZIRC`.
Also, after concatenation, the gathered data's channel is set to 0 (channel does not have consistent meaning).

## Related links

doc: https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-architecture/sensing/data-types/point-cloud/#recommended-processing-pipeline

## Tests performed
Before
``` Text
...
fields:
- name: x
  offset: 0
  datatype: 7
  count: 1
- name: y
  offset: 4
  datatype: 7
  count: 1
- name: z
  offset: 8
  datatype: 7
  count: 1
- name: intensity
  offset: 12
  datatype: 7
  count: 1
is_bigendian: false
point_step: 16
...
```

After
``` text
...
fields:
- name: x
  offset: 0
  datatype: 7
  count: 1
- name: y
  offset: 4
  datatype: 7
  count: 1
- name: z
  offset: 8
  datatype: 7
  count: 1
- name: intensity
  offset: 12
  datatype: 7
  count: 1
- name: padding
  offset: 16
  datatype: 2
  count: 1
- name: return_type
  offset: 17
  datatype: 2
  count: 1
- name: channel
  offset: 18
  datatype: 4
  count: 1
is_bigendian: false
point_step: 20
...
```

## Notes for reviewers

I only checked the sensing part is working, please check the other parts if it's needed.

## Interface changes

N/A

## Effects on system behavior

N/A

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/autoware_point_types/include/autoware_point_types/types.hpp [hunk 1]
```diff
@@ -54,6 +54,29 @@ enum ReturnType : uint8_t {
...
+{
+  float x{0.0F};
+  float y{0.0F};
+  float z{0.0F};
+  float intensity{0.0F};
+  union {  // for memory alignment
+    uint16_t _data;
+    struct
+    {
+      uint8_t padding{0U};
```
Comment (a-maumau) [inline common/autoware_point_types/include/autoware_point_types/types.hpp:67]: for all reviewers, is this `padding` OK to be the field name?

**Comment by badai-nguyen** [comment]
@xmfcx I remember you mentioned [this issue in S/P WG ](https://github.com/orgs/autowarefoundation/discussions/4633). Do you have any comment? 
I think you prososed idea to keep the same size of pointcloud, did you?

**Comment by xmfcx** [comment]
The intensity should be 1 byte and you don't need any paddings.

See here:

https://github.com/autowarefoundation/autoware-documentation/blob/01e6b75fe73b1831aba61d52d165c4f67eedd7d7/docs/design/autoware-architecture/sensing/data-types/point-cloud.md#point-cloud-fields

This is how it should be.

Then this pr was merged because the point types didn't match the repo:
- https://github.com/autowarefoundation/autoware-documentation/pull/478

But we agreed to use 1 byte once we fixed the repo.

But I believe this should start getting fixed from the nebula driver, the sensing layer, then the concatenator.

Even in the latest documentation page, you can see how 1 byte is enough to represent the intensity: https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-architecture/sensing/data-types/point-cloud/#intensity

**Comment by xmfcx** [comment]
Before: 4 * 4 bytes; (x, y, z, i) : (f32, f32, f32, f32)
After: 3 * 4 bytes + 2 * 1 byte + 1 * 2 byte (x, y, z, i, r, c) : (f32, f32, f32, ui8, ui8, ui16)

Same size, same bandwidth, perfect padding. Even if you don't need the fields, same speed.

**Comment by a-maumau** [comment]
I've made the commit fixing the intensity type mentioned by @xmfcx in the comment: changed the intensity from `float` to `uint8`.
Since the code assumed the intensity of input to be `float`, it requires updating the casts in the ring_outlier_filter_nodelet.cpp after modifying other parts (e.g., the nebula driver).

**Comment by a-maumau** [comment]
https://github.com/autowarefoundation/autoware.universe/pull/6996 overlaps with this PR so I will close this PR.

**Pull request closed without merging**